jerry-port/default/default-date.c

```diff
@@ -21,33 +24,32 @@
 #include "jerryscript-port-default.h"
 
 /**
- * Default implementation of jerry_port_get_time_zone. Uses 'gettimeofday' if
- * available on the system, does nothing otherwise.
+ * Default implementation of jerry_port_get_local_tza. Uses the 'tm_gmtoff' field
+ * of 'struct tm' (a GNU extension) filled by 'localtime_r' if available on the
+ * system, does nothing otherwise.
  *
- * @return true - if 'gettimeofday' is available and executed successfully,
- *         false - otherwise.
+ * @return offset between UTC and local time at the given unix timestamp, if
+ *         available. Otherwise, returns 0, assuming UTC time.
  */
-bool jerry_port_get_time_zone (jerry_time_zone_t *tz_p) /**< [out] time zone structure to fill */
+double jerry_port_get_local_tza (double js_time_ms,  /**< ms since unix epoch */
+                                 bool is_utc)  /**< is the time above in UTC? */
 {
-#ifdef __GNUC__
-  struct timeval tv;
-  struct timezone tz;
-
-  /* gettimeofday may not fill tz, so zero-initializing */
-  tz.tz_minuteswest = 0;
-  tz.tz_dsttime = 0;
-
-  if (gettimeofday (&tv, &tz) == 0)
+#ifdef HAVE_TM_GMTOFF
+  struct tm tm;
+  time_t now = (time_t) (js_time_ms / 1000);
+  localtime_r (&now, &tm);
```

Comment:
A little bit of thinking aloud: `localtime_r` is not C99 but an extension defined by the POSIX standard. Which means that this code will not necessarily compile on all targets. Unless the existence of the `tm.tm_gmtoff` GNU extension implies the POSIX extension as well.

Another approach could be to use `timegm(&tm) - t` instead of `tm.tm_gmtoff` but `timegm` is also a GNU/BSD extension. (Which could be emulated if not present on the system, but I'm not sure it would be worth the effort.)

So, it seems to me that we need to rely on at least two extensions ot get this work. I'm not sure yet which way to go. (That's way I was thinking aloud. I'm open to inputs.)

Reply:
My point was mainly that if a given platform has `tm.tm_gmtoff` then it _most probably_ has `localtime_r` as well. But maybe there is a better way that has support on more platforms, I'm open to suggestions. We can always change / improve this later, however.

Reply:
Btw what about windows? Does gcc on windows support this?

Reply:
Correct, this is the default implementation. Most embedded targets won't even use this, or they will just use the `return 0;` default implementation, which should have the same "assume UTC" behaviour that they always had.

Windows does not support `tm.tm_gmtoff`. It's also not supported by mingw. However, it's fairly straightforward to shim it by using `GetTimeZoneInformationForYear` if someone wants to do this at some point. I'd rather not do it in this PR, as it's already gotten quite big (and it's a PITA for me to test on windows...)